mllib/src/main/scala/org/apache/spark/ml/recommendation/ALS.scala

```diff
@@ -82,12 +82,20 @@ private[recommendation] trait ALSModelParams extends Params with HasPredictionCo
    * Attempts to safely cast a user/item id to an Int. Throws an exception if the value is
    * out of integer range.
    */
-  protected val checkedCast = udf { (n: Double) =>
-    if (n > Int.MaxValue || n < Int.MinValue) {
-      throw new IllegalArgumentException(s"ALS only supports values in Integer range for columns " +
-        s"${$(userCol)} and ${$(itemCol)}. Value $n was out of Integer range.")
-    } else {
-      n.toInt
+  protected val checkedCast = udf { (n: Any) =>
+    n match {
+      case v: Int => v // Avoid unnecessary casting
+      case v: Number =>
+        val intV = v.intValue()
+        // Checks if number within Int range and has no fractional part.
+        if (v.doubleValue == intV) {
```

Comment:
I think the test does work, but may be slightly more direct to check if `v.doubleValue % 1.0 == 0.0`? this also works.